### PR TITLE
Add tag on cluster

### LINKED ISSUE #183: Handle error like (500 internal server error) from some api call  (not all) during creation and deletion of cluster
### Explain problem to solve

If you get an error like (like (500 internal server error))  for some api call  (Not all)  during the creation of the cluster the first time , the controller will not store the id of the ressource  already created, and will recreate a new cluster.

### Describe the solution you would like

Option1: 
* Read tag are already here
* We will use the name of each component because it is a unique tag with cluster uuid.

Option2:
* Handle it like throttling

We choose Option1.

### Additional context

None

### Environment

```markdown
- Kubernetes version: (use `kubectl version`): v1.22.5-rancher1-1
- OS (e.g. from `/etc/os-release`): CentOS Linux 7 (Core)
- Kernel (e.g. `uname -a`): 3.10.0-1160.15.2.el7.x86_64
- cluster-api-provider-outscale version: v1.1.3
- cluster-api version: current
- Install tools: calico, ingress-nginx, osc-ccm
- Kubernetes Distribution: rke
- Kubernetes Distribution version: v1.3.4
```

### PR DESCRIPTION
**Pull request by outscale-vbr**
**What type of PR is this?**
/kind feature 

**What this PR does / why we need it**:
Use tag name not to have duplicate cluster if the api is down during the first initialization.
Closes #183 

**Which issue(s) this PR fixes**:
Fixes #183 

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Comment by outscale-vbr** [comment]
@outscale-mdr @outscale-hmi Are you ok with this PR ?

File: cloud/tag/tag.go [hunk 1]
```diff
@@ -27,8 +27,13 @@ import (
...
 )
 
+//go:generate ../../bin/mockgen -destination mock_tag/tag_mock.go -package mock_tag -source ./tag.go
+type OscTagInterface interface {
+	ReadTag(tagKey string, tagValue string) (*osc.Tag, error)
+}
+
 // AddTag add a tag to a resource
-//func AddTag(tagKey string, tagValue string, resourceIds []string, api *osc.APIClient, auth context.Context) (error, *http.Response) {
+// func AddTag(tagKey string, tagValue string, resourceIds []string, api *osc.APIClient, auth context.Context) (error, *http.Response) {
```
Comment (outscale-hmi) [inline cloud/tag/tag.go:36]: you can remove comment

**Review by outscale-hmi — COMMENTED** [review]
Good for me

File: cloud/tag/tag.go (hunk 1 above)
Comment (outscale-vbr) [reply cloud/tag/tag.go]: Ok remove

File: hack/boilerplate/boilerplate.py [hunk 2]
```diff
@@ -149,7 +148,7 @@ def file_extension(filename):
 
 skipped_dirs = ['Godeps', 'third_party', '_gopath', '_output', '.git', 'cluster/env.sh',
                 "vendor", "test/e2e/generated/bindata.go", "hack/boilerplate/test",
-                "pkg/kubectl/generated/bindata.go", "tilt_modules", "cloud/services/net/mock_net", "cloud/services/security/mock_security", "cloud/services/service/mock_service", "cloud/services/storage/mock_storage", "cloud/services/compute/mock_compute", "api/v1beta1/zz_generated.deepcopy.go"]
+                "pkg/kubectl/generated/bindata.go", "tilt_modules", "cloud/services/net/mock_net", "cloud/services/security/mock_security", "cloud/services/service/mock_service", "cloud/services/storage/mock_storage", "cloud/services/compute/mock_compute", "cloud/tag/mock_tag", "api/v1beta1/zz_generated.deepcopy.go"]
```
Comment (outscale-hmi) [inline hack/boilerplate/boilerplate.py:151]: format

Comment (outscale-vbr) [reply hack/boilerplate/boilerplate.py:151]: @outscale-hmi  What do you mean ?

Comment (outscale-vbr) [reply hack/boilerplate/boilerplate.py:151]: All in one line ?

Comment (outscale-vbr) [reply hack/boilerplate/boilerplate.py]: Ok done

**Review by outscale-hmi — APPROVED** [review]
OK

**Pull request merged**